### DISCUSSION TITLE
Custom navigation-tree with limited number of pages?

### DISCUSSION THREAD

**Discussion by flohbus** [datenstrom/community#293] [Ask a question] [answered]
Hi,

not sure if that is actually an issue or if I am missing some config opportunity. I am using the blog plugin as well as the navigation-tree and it occurred to me, that for the blog part, the navigation-tree adds a menu for the "blogsite" page and submenu entries for each "blog" page, speak: eache blog post. I found this quite odd and would have preferred to only see the main blog page in the menu but no submenu. I do not actually need 50 submenu bullets for 50 blos posts... (actually I have not tested with more than 5 so I am not sure if it would actually display 50).
To work around this, I have modified the navigation-tree snippet to only construct a sub menu, if the page-tempalte for the main menu entry is "default" (could also be: is NOT "blogsite").
Is that something worth a fix/enhancement or did I just not get it right? :-)

I am new to yellow, so might not oversee the whole thing yet, but so far I like it very much.

Thanks,
Georg

**Comment by BsNoSi** [comment]
If you add a navigation with automatic creation of a folder tree, then this is the behaviour as expected.

What do you want to achieve? 

The blog install (from the original setup) creates the basic structure for a blog without submenues. If you want to create a page with a blog and some additional pages, that contain subpages and this subpages shall have a submenu, you have to create a template „blog“ that is without submenu creation and you have to assign it to each blog page:

```
---
title: blogpage
template: blog
---
```
Or you design a template „outhers“ and assign this to each other page and leave the „default“ template. used by the blog pages, as is. 

Another option is using the sidbare for submenu. In static areas you add a submenu, in the blog area, you can add blog functions ([see description](https://developers.datenstrom.se/de/help/how-to-make-a-blog)).

**Comment by markseuffert** [comment, marked as answer]
Hi Georg, you did it exactly as it was intended. If the existing navigation doesn't fit your needs you customise it. The `navigation-tree` shows all pages. So you're right, for a blog that doesn't make a lot of sense. You can limit the entries in the navigation or show only the latest entries. 

Here's an example with the latest 10 blog entries:

```
<?php list($name, $pages, $level) = $yellow->getSnippetArgs() ?>
<?php if(!$pages) $pages = $yellow->pages->top() ?>
<?php $yellow->page->setLastModified($pages->getModified()) ?>
<?php if(!$level): ?>
<div class="navigation-tree">
<?php endif ?>
<ul>
<?php foreach($pages as $page): ?>
<?php $children = $page->getChildren() ?>
<?php if($page->get("template")=="blogpages") $children->sort("published")->reverse()->limit(10) ?>
<li><a<?php echo $page->isActive() ? " class=\"active\"" : "" ?> href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("titleNavigation") ?></a><?php if($children->count()) { echo "\n"; $yellow->snippet($name, $children, $level+1); } ?></li>
<?php endforeach ?>
</ul>
<?php if(!$level): ?>
</div>
<div class="navigation-banner"></div>
<?php endif ?>
```
See also https://datenstrom.se/yellow/help/

**Comment by BsNoSi** [comment]
I wonder why you are using „navigation-tree“ instead of „navigation“, together with some functions of the blog:

```
#### Tags

[blogtags /]

#### Last

[blogrecent / 3]

### Similar

[blogrelated / 3]

#### Almanach

[2016](/published:2016) • [2015](/published:2015)

[2014](/published:2014) • [2013](/published:2013) 
```

You can place this into the „sidebar“ of the blog folder and use the `blog`-template.

Or are you combining blog pages with static pages, and some of them shall have a sub menu?

For this case, I created for a page a plugin „index“, that lists all files in a dedicated directory. It is in beta-state currently, but if you are interested, I may upload it to github.
